client/apollo/react/src/AccordionCore/AccordionCoreCommon.tsx

```diff
@@ -0,0 +1,66 @@
+import {
+  type ComponentProps,
+  type MouseEvent,
+  type ReactNode,
+  useCallback,
+  useRef,
+} from "react";
+import keyboardUp from "@material-symbols/svg-400/rounded/keyboard_arrow_up-fill.svg";
+import { Svg } from "../Svg/Svg";
+
+type AccordionProps = {
+  summary: ReactNode;
+  children: ReactNode;
+  isOpen?: boolean;
+  summaryProps?: ComponentProps<"summary">;
+  onClick?: (event: MouseEvent<HTMLDetailsElement>) => void;
+} & Partial<ComponentProps<"details">>;
+
+export const AccordionCore = ({
+  summary,
+  children,
+  className,
+  summaryProps,
+  isOpen = false,
+  onClick,
+  ...detailsProps
+}: AccordionProps) => {
+  const detailsRef = useRef<HTMLDetailsElement>(null);
+
+  const handleToggle = useCallback(
+    (event: MouseEvent<HTMLDetailsElement>) => {
+      if (onClick) {
+        event.preventDefault();
+        onClick(event);
+      }
+    },
+    [onClick],
+  );
+
+  return (
+    <details
+      className={["af-apollo-accordion", className].filter(Boolean).join(" ")}
+      ref={detailsRef}
+      open={isOpen}
+      {...detailsProps}
+    >
+      <summary
+        onClick={handleToggle}
+        role="button"
+        tabIndex={0}
+        {...summaryProps}
+        className={["af-apollo-accordion__summary", summaryProps?.className]
+          .filter(Boolean)
+          .join(" ")}
+      >
+        {summary}
+        <div className={["af-accordion__arrow", "af-click-icon"].join(" ")}>
+          <div className="af-icon af-icon--primary">
+            <Svg src={keyboardUp} role="presentation" />
+          </div>
```

Comment:
Pourquoi faire cela plutôt que d'utiliser le composant Icon ?

Reply:
C'est la recommandation de Paul, voir commentaire en haut.

Reply:
Oui, il suggère d'utiliser une icône avec une div parent avec les classes du ClickIcon plutôt qu'un ClickIcon directement, et je suis totalement d'accord avec lui.
La différence est que là tu n'utilises pas les classes de ClickIcon mais celle d'Icon.
Mais la meilleure solution selon moi est de faire la suggestion de Paul mais avec un Icon à la place de Svg (car Svg est déprécié contrairement à Icon + c'est comme cela que c'est fait dans ClickIcon) comme ceci : 
```suggestion
          <div className="af-icon af-icon--primary">
            <Icon src={keyboardUp} role="presentation" />
          </div>
```

Reply:
d'accord